### PR TITLE
Add Border Margins to View Components

### PR DESCRIPTION
**Pull request by natashaU**
This adds support for the following primer classes to `classify.rb`:

```
"border-top-0" ----------->  border_top: 0
"border-bottom-0"--------> border_bottom: 0
"border-left-0" ------------> border_left: 0
"border-right-0" -----------> border_right: 0

```

**cc/ @joelhawksley @aellispierce**

### CODE REVIEW

File: test/primer/classify_test.rb [hunk 1]
```diff
@@ -167,11 +167,15 @@ def test_col
...
-    assert_generated_class("border-top",     { border: :top })
-    assert_generated_class("border-bottom",  { border: :bottom })
-    assert_generated_class("border-right",   { border: :right })
-    assert_generated_class("border-y",       { border: :y })
+    assert_generated_class("border-left",     { border: :left })
+    assert_generated_class("border-top",      { border: :top })
+    assert_generated_class("border-bottom",   { border: :bottom })
+    assert_generated_class("border-right",    { border: :right })
+    assert_generated_class("border-y",        { border: :y })
+    assert_generated_class("border-top-0",    { border: :top_0 })
```
Comment (joelhawksley) [inline test/primer/classify_test.rb:175]: I _think_ this should be `border_top: 0`, to match Styled System: https://styled-system.com/table/#border. I'd usually ask @emplums, but she is OOO until Monday 😄

Comment (natashaU) [reply test/primer/classify_test.rb]: @joelhawksley Just pushed up new changes to change it to border_top: 0

File: lib/primer/classify.rb [hunk 2]
```diff
@@ -77,7 +77,7 @@ class Classify
         ]
       }
     }.freeze
-    BORDER_KEYS = [:border, :border_color].freeze
+    BORDER_KEYS = [:border, :border_color, :border_top, :border_bottom, :border_left, :border_right].freeze
```
Comment (joelhawksley) [inline lib/primer/classify.rb:80]: Per Zoom: Split into two constants ❤️

File: lib/primer/classify.rb [hunk 3]
```diff
@@ -170,6 +170,8 @@ def extract_hash(styles_hash)
             end
 
             dasherized_val = val.to_s.dasherize
+            dasherized_key = key.to_s.dasherize
```
Comment (joelhawksley) [inline lib/primer/classify.rb:173]: Let's inline this below, as it's only used once 😄

**Comment by joelhawksley** [comment]
Might we update the README with this behavior?

**Comment by natashaU** [comment]
> Might we update the README with this behavior?

Added it here @joelhawksley: https://github.com/primer/view_components/pull/70#issuecomment-689837143

CC/ @emplums

**Review by joelhawksley — APPROVED** [review]

**Comment by joelhawksley** [comment]
Mind adding a changelog entry to this PR?

**Comment by natashaU** [comment]
> Mind adding a changelog entry to this PR?

@joelhawksley Added a changelog entry and README! Thanks for the feedback 🎉

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,9 @@
+# main
+
+* Added support for border margins, such as: `border_top: 0`
```
Comment (joelhawksley) [inline CHANGELOG.md:3]:
```suggestion
* Add support for border margins, such as: `border_top: 0`.
```

**Pull request merged**